nostrpostrlib/src/main/java/nostr/postr/Utils.kt

```diff
@@ -81,4 +81,11 @@ fun Int.toByteArray(): ByteArray {
 }
 
 fun ByteArray.toNsec() = Bech32.encodeBytes(hrp = "nsec", this, Bech32.Encoding.Bech32)
-fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
\ No newline at end of file
+fun ByteArray.toNpub() = Bech32.encodeBytes(hrp = "npub", this, Bech32.Encoding.Bech32)
+
+fun String.rawHexForm() = Bech32.decodeBytes(this).second
```

Comment:
Does rawHexForm() return a byte array? Why not name it

```suggestion
fun String.bechToBytes() = Bech32.decodeBytes(this).second
```

or if the hrp is ignored, we could put in some safe-guards by at least making it easy to fail if the user put in his nsec where we expected an npub:

```suggestion
fun String.bechToBytes(hrp:String?=null): ByteArray {
    val decoded = Bech32.decodeBytes(this)
    hrp?.also { 
        if (it !== decoded.first) {
            throw IllegalArgumentException("Expected $it but was ${decoded.first}")
        }
    }
    return decoded.second
}
```

Reply:
Will adjust it then.:+1: